.github/workflows/entropy-check.yml

```diff
@@ -0,0 +1,46 @@
+---
+name: entropy-check
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    permissions: write-all
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Install python env
+        uses: ./.github/actions/install-python-env
+
+      - name: Process repository entropy
+        run: poetry run almanack process_repo_entropy . > entropy_report.txt
+
+      - name: Format Entropy Report
+        id: format_report
+        run: |
+          CONTENT=$(sed 's/"/\\"/g' entropy_report.txt | awk '{print}' ORS='\n')
+          {
+            echo "REPORT_CONTENT<<EOF";
+            echo "$CONTENT";
+            echo "EOF";
+          } >> "$GITHUB_ENV"
```

Comment:
Consider modifying this block to implement the use of a code block through markdown if it helps with the display any.